src/link/sdas_obj.cpp

```diff
@@ -223,6 +223,7 @@ void sdobj_ReadFile(FileStackNode const &where, FILE *file, std::vector<Symbol>
 
 	getToken(nullptr, "'H' line is too short");
 	uint32_t expectedNbSymbols = parseNumber(where, lineNo, token, numberType);
+	fileSymbols.reserve(expectedNbSymbols);
```

Comment:
I know that this technically guarantees `fileSymbols` won't reallocate with `expectedNbSymbols` or fewer items, but it's not nonlocally obvious. In some parts of the code we use `std::deque` for collections that shouldn't move their items; in others, we have collections of indexes or pointers; and a third alternative would be to error if we exceed the expected number of symbols.

My preference would be for a `std::deque` here; it's a simpler change than the others. (And as we discussed, it seems *possible* that some weird SDCC object could have more symbols than expected but still ought to work.)

Reply:
It is anything but a simple change, since the symbols get passed around all the time, and this needs to be homogeneous with RGBDS objects too. So this would change a *lot* of function signatures, and that's starting to overflow the scope of the PR.

I'll make the warning into an error.